### PR TITLE
Allow users to rescind authorization to access build api

### PR DESCRIPTION
**Pull request by jemoreira**

### CODE REVIEW

File: pkg/app/app.go [hunk 1]
```diff
@@ -310,6 +301,98 @@ func (c *App) parseAuthorizationResponse(r *http.Request) (string, error) {
...
+		OAuth2State: randomHexString(),
+	}
+	if err := a.setOrUpdateSession(w, &s); err != nil {
+		return err
+	}
+	_, err := fmt.Fprintf(w, pageTemplate, s.OAuth2State)
+	return err
+}
+
+func (a *App) RescindAuthorization(w http.ResponseWriter, r *http.Request, user accounts.User) error {
```
Comment (ser-io) [inline pkg/app/app.go:329]: nit: `RescindAuthorizationHandler`?

File: pkg/app/app.go [hunk 2]
```diff
@@ -310,6 +301,98 @@ func (c *App) parseAuthorizationResponse(r *http.Request) (string, error) {
...
+	r.ParseForm()
+	stateSlice, ok := r.PostForm["csrf_token"]
+	if !ok || len(stateSlice) == 0 {
```
Comment (ser-io) [inline pkg/app/app.go:332]: should we add unit test to make sure this security check does not get lost in future refactoring?

**Review by ser-io — CHANGES_REQUESTED** [review]

File: pkg/app/app.go (hunk 1 above)
Comment (jemoreira) [reply pkg/app/app.go]: Done

File: pkg/app/app.go (hunk 2 above)
Comment (jemoreira) [reply pkg/app/app.go:332]: Done

File: pkg/app/app.go [hunk 3]
```diff
@@ -310,6 +301,98 @@ func (c *App) parseAuthorizationResponse(r *http.Request) (string, error) {
...
+	}
+	defer func() {
+		if err := a.databaseService.DeleteBuildAPICredentials(user.Username()); err != nil {
+			log.Printf("Failed to delete credentials from database: %v", err)
+		}
+	}()
+	if err := a.oauth2Helper.Revoke(tk); err != nil {
+		return err
+	}
+	fmt.Fprintln(w, "Authorization revoked")
```
[the remark this answers is not in the document]
Comment (jemoreira) [reply pkg/app/app.go]: Done

**Review by ser-io — APPROVED** [review]

**Pull request merged**